compiler/rustc_codegen_ssa/src/target_features.rs

```diff
@@ -210,6 +210,34 @@ const WASM_ALLOWED_FEATURES: &[(&str, Option<Symbol>)] = &[
     ("nontrapping-fptoint", Some(sym::wasm_target_feature)),
 ];
 
+const XTENSA_ALLOWED_FEATURES: &[(&str, Option<Symbol>)] = &[
+    ("fp", Some(sym::xtensa_target_feature)),
+    ("windowed", Some(sym::xtensa_target_feature)),
+    ("bool", Some(sym::xtensa_target_feature)),
+    ("loop", Some(sym::xtensa_target_feature)),
+    ("sext", Some(sym::xtensa_target_feature)),
+    ("nsa", Some(sym::xtensa_target_feature)),
+    ("mul32", Some(sym::xtensa_target_feature)),
+    ("mul32high", Some(sym::xtensa_target_feature)),
+    ("div32", Some(sym::xtensa_target_feature)),
+    ("mac16", Some(sym::xtensa_target_feature)),
+    ("dfpaccel", Some(sym::xtensa_target_feature)),
+    ("s32c1i", Some(sym::xtensa_target_feature)),
+    ("threadptr", Some(sym::xtensa_target_feature)),
+    ("extendedl32r", Some(sym::xtensa_target_feature)),
+    ("atomctl", Some(sym::xtensa_target_feature)),
+    ("memctl", Some(sym::xtensa_target_feature)),
+    ("debug", Some(sym::xtensa_target_feature)),
+    ("exception", Some(sym::xtensa_target_feature)),
+    ("coprocessor", Some(sym::xtensa_target_feature)),
+    ("interrupt", Some(sym::xtensa_target_feature)),
+    ("rvector", Some(sym::xtensa_target_feature)),
+    ("timerint", Some(sym::xtensa_target_feature)),
+    ("prid", Some(sym::xtensa_target_feature)),
+    ("regprotect", Some(sym::xtensa_target_feature)),
+    ("miscsr", Some(sym::xtensa_target_feature)),
+];
```

Comment:
Do these need to added

```Rust
    ("density", Some(sym::xtensa_target_feature)),
    ("highpriinterrupts", Some(sym::xtensa_target_feature)),
```

Reply:
No to both I think.

- "density" does not affect register availablity
- "highpriinterrupts" is enabled via "exception" depending on the processor.

Reply:
Is there a possibility that they may be required by future code?